src/core/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,817 @@
+  ).toBeTruthy();
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(1);
+});
+
+// available track segments 0--10--20--30--40--100
+it('should render odd tracks based on even number of values and `auto` trackDisplayMode', () => {
+  const { container } = render(<Slider values={[10, 20, 30, 40]} />);
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  // segments 10-20, 30-40
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(2);
+});
+
+it('should render 3 `even-segments` 0-10,20-30,40-100', () => {
+  const { container } = render(
+    <Slider trackDisplayMode='even-segments' values={[10, 20, 30, 40]} />,
+  );
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  // segments 0-10,20-30,40-100
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(3);
+});
+
+it('should render 2 `odd-segments` 10-20, 30-40', () => {
+  const { container } = render(
+    <Slider trackDisplayMode='odd-segments' values={[10, 20, 30, 40]} />,
+  );
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(2);
+});
+
+it('should not render track', () => {
+  const { container } = render(
+    <Slider trackDisplayMode='none' values={defaultSingleValue} />,
+  );
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(0);
+});
+
+it('should activate thumb on pointerDown', () => {
+  const wrapper = render(<Slider values={defaultSingleValue} />);
+  const { container } = wrapper;
+  assertBaseElement(container);
+  const thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb).toBeTruthy();
+  expect(thumb.classList.contains('iui-active')).toBeFalsy();
+
+  act(() => {
+    fireEvent.pointerDown(thumb, {
+      pointerId: 5,
+      buttons: 1,
+      clientX: 510,
+    });
+  });
+  expect(thumb.classList.contains('iui-active'));
+});
+
+it('should process keystrokes when thumb has focus', () => {
+  const handleOnChange = jest.fn();
+  const wrapper = render(
+    <Slider values={[50]} step={5} setFocus onChange={handleOnChange} />,
+  );
+  const { container } = wrapper;
+  assertBaseElement(container);
+  let thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb).toBeTruthy();
+  expect(thumb.classList.contains('iui-active')).toBeFalsy();
+
+  expect(document.activeElement).toEqual(thumb);
+  expect(thumb.getAttribute('aria-valuenow')).toEqual('50');
+
+  act(() => {
+    fireEvent.keyDown(thumb, { key: 'ArrowLeft' });
+  });
+  thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
```

Comment:
Do you actually need to query for thumb repeatedly? Does original thumb reference not update its attributes?

Reply:
Similar comment: https://github.com/iTwin/iTwinUI-react/pull/197/files#r672185321

Reply:
It is not!!! I got stuck on these ones..